data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessorConfig.java

```diff
@@ -5,50 +5,90 @@
 import jakarta.validation.constraints.NotNull;
 import jakarta.validation.constraints.Size;
+import org.opensearch.dataprepper.model.configuration.PluginModel;
 import org.opensearch.dataprepper.plugins.lambda.common.config.AwsAuthenticationOptions;
 import org.opensearch.dataprepper.plugins.lambda.common.config.BatchOptions;
-import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.BATCH_EVENT;
+import org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig;
+import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.DEFAULT_CONNECTION_RETRIES;
+import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.DEFAULT_SDK_TIMEOUT;
 import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.REQUEST_RESPONSE;
 
-public class LambdaProcessorConfig {
+import javax.annotation.Nullable;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
 
-    private static final int DEFAULT_CONNECTION_RETRIES = 3;
+public class LambdaProcessorConfig {
+    //Ensures 1:1 mapping of events input to lambda and response from lambda
+    public static final String STRICT = "strict";
+    //When #input events to lambda is not equal to the response from lambda
+    public static final String AGGREGATE = "aggregate";
 
     @JsonProperty("aws")
     @NotNull
     @Valid
     private AwsAuthenticationOptions awsAuthenticationOptions;
 
+    @JsonPropertyDescription("Lambda Function Name")
     @JsonProperty("function_name")
     @NotEmpty
     @Size(min = 3, max = 500, message = "function name length should be at least 3 characters")
     private String functionName;
 
+    @JsonPropertyDescription("Total retries we want before failing")
     @JsonProperty("max_retries")
     private int maxConnectionRetries = DEFAULT_CONNECTION_RETRIES;
 
+    @JsonPropertyDescription("invocation type defines the way we want to call lambda function")
     @JsonProperty("invocation_type")
     private String invocationType = REQUEST_RESPONSE;
 
-    @JsonProperty("payload_model")
-    private String payloadModel = BATCH_EVENT;
+    @JsonPropertyDescription("sdk timeout defines the time sdk maintains the connection to the client before timing out")
+    @JsonProperty("sdk_timeout")
+    private Duration sdkTimeout = DEFAULT_SDK_TIMEOUT;
+
+    @JsonPropertyDescription("mode defines the way dataprepper treats the response from lambda")
+    @JsonProperty("response_processing_mode")
+    private String responseProcessingMode = STRICT;
 
     @JsonProperty("batch")
     private BatchOptions batchOptions;
 
+    @JsonPropertyDescription("defines a condition for event to use this processor")
     @JsonProperty("lambda_when")
     private String whenCondition;
 
+    @JsonPropertyDescription("Codec configuration for parsing Lambda responses")
+    @JsonProperty("response_codec")
+    @Valid
+    @Nullable
```

Comment:
This should not be `@Nullable`. It should have be required and have a default value.